src/debuginfo.cpp

```diff
@@ -881,40 +867,128 @@ static void jl_getDylibFunctionInfo(char **name, char **filename, size_t *line,
+        // record source file name and line number
+        if (frame_info_line.FileName && filename)
+            jl_copy_str(filename, frame_info_line.FileName);
+        *line = frame_info_line.LineNumber;
+    }
+    else if (*fromC) {
+        // No debug info, use dll name instead
+        if (filename)
+            jl_copy_str(filename, fname);
+    }
+    jl_in_stackwalk = 0;
 #endif
-        lookup_pointer(context, name, line, filename, inlinedat_line, inlinedat_file, pointer+slide,
-                       fbase == jl_sysimage_base, fromC);
-        if (jl_sysimage_base == fbase && sysimg_fvars) {
+    const object::ObjectFile *object;
+    llvm::DIContext *context;
+    bool isSysImg;
+    void *saddr;
+    int64_t slide;
+    if (!jl_dylib_DI_for_fptr(pointer, &object, &context, &slide, skipC, &isSysImg, &saddr, name, filename)) {
+        *fromC = 1;
+        return;
+    }
+
+    lookup_pointer(context, name, line, filename, inlinedat_line, inlinedat_file, pointer+slide,
+                   isSysImg, fromC);
+    if (isSysImg && sysimg_fvars) {
 #ifdef _OS_LINUX_
-            unw_proc_info_t pip;
-            if (!saddr && unw_get_proc_info_by_ip(unw_local_addr_space,
-                                                  pointer, &pip, NULL) == 0)
-                saddr = (void*)pip.start_ip;
-#endif
-            if (saddr) {
-                for (size_t i = 0; i < sysimg_fvars_n; i++) {
-                    if (saddr == sysimg_fvars[i]) {
-                        *outer_linfo = sysimg_fvars_linfo[i];
-                        break;
-                    }
+        unw_proc_info_t pip;
+        if (!saddr && unw_get_proc_info_by_ip(unw_local_addr_space,
+                                              pointer, &pip, NULL) == 0)
+            saddr = (void*)pip.start_ip;
+#endif
+        if (saddr) {
+            for (size_t i = 0; i < sysimg_fvars_n; i++) {
+                if (saddr == sysimg_fvars[i]) {
+                    *outer_linfo = sysimg_fvars_linfo[i];
+                    break;
                 }
             }
         }
     }
-    else {
-        *fromC = 1;
+}
+
+int jl_DI_for_fptr(uint64_t fptr, uint64_t *symsize, int64_t *slide, const object::ObjectFile **object,
+#ifdef USE_MCJIT
+                      llvm::DIContext **context
+#else
+                      std::vector<JITEvent_EmittedFunctionDetails::LineStart> *lines
+#endif
+                      )
+{
+    int found = 0;
+#ifndef USE_MCJIT
+    std::map<size_t, FuncInfo, revcomp> &fmap = jl_jit_events->getMap();
+    std::map<size_t, FuncInfo, revcomp>::iterator fit = fmap.find(fptr);
+
+    if (fit != fmap.end()) {
+        *symsize = fit->second.lengthAdr;
+        *lines = fit->second.lines;
+        *slide = 0;
+        found = 1;
     }
+#else // MCJIT version
+    std::map<size_t, ObjectInfo, revcomp> &objmap = jl_jit_events->getObjectMap();
+    std::map<size_t, ObjectInfo, revcomp>::iterator fit = objmap.find(fptr);
```

Comment:
since you aren't putting fptr's in this map anymore, `find` isn't going to find them in here